File: internal/content/testdata/bench/csharp/nc-valid-parentheses.cs
using System.Collections.Generic;

public class Solution {
    public bool is_valid(string s) {
        var stack = new Stack<char>();
        foreach (char ch in s) {
            if (ch == ')' || ch == ']' || ch == '}') {
                if (stack.Count == 0) return false;
                char top = stack.Pop();
                if (ch == ')' && top != '(') return false;
                if (ch == ']' && top != '[') return false;
                if (ch == '}' && top != '{') return false;
            } else {
                stack.Push(ch);
            }
        }
        return stack.Count == 0;
    }
}

File: internal/content/testdata/bench/java/nc-valid-parentheses.java
import java.util.*;
class Solution {
    public boolean is_valid(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        for (char ch : s.toCharArray()) {
            if (ch == ')' || ch == ']' || ch == '}') {
                if (stack.isEmpty()) return false;
                char top = stack.pop();
                if (ch == ')' && top != '(') return false;
                if (ch == ']' && top != '[') return false;
                if (ch == '}' && top != '{') return false;
            } else {
                stack.push(ch);
            }
        }
        return stack.isEmpty();
    }
}
